plugins/inputs/irqstat/irqstat.go

```diff
@@ -0,0 +1,128 @@
+package irqstat
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"log"
+	"strconv"
+	"strings"
+)
+
+type Irqstat struct {
+	Include []string
+}
+
+type IRQ struct {
+	ID     string
+	Fields map[string]interface{}
+	Tags   map[string]string
+}
+
+func NewIRQ(id string) *IRQ {
+	return &IRQ{ID: id, Fields: make(map[string]interface{}), Tags: make(map[string]string)}
+}
+
+const sampleConfig = `
+  ## A list of IRQs to include for metric ingestion, if not specified
+  ## will default to collecting all IRQs.
+  # include = ["0", "1", "30", "NET_RX"]
+`
+
+func (s *Irqstat) Description() string {
+	return "This plugin gathers interrupts data from /proc/interrupts and /proc/softirqs."
+}
+
+func (s *Irqstat) SampleConfig() string {
+	return sampleConfig
+}
+
+func parseInterrupts(irqdata string, include []string) []IRQ {
+	var irqs []IRQ
+	var cpucount int
+	scanner := bufio.NewScanner(strings.NewReader(irqdata))
+	for scanner.Scan() {
+		var irqval, irqtotal int64
+		var irqtype, irqdevice string
+		fields := strings.Fields(scanner.Text())
+		ff := fields[0]
+		if ff == "CPU0" {
+			cpucount = len(fields)
+		}
+
+		if ff[len(ff)-1:] == ":" {
+			fields = fields[1:len(fields)]
+			irqid := ff[:len(ff)-1]
+			irq := NewIRQ(irqid)
+			_, err := strconv.ParseInt(irqid, 10, 64)
+			if err == nil {
+				irqtype = fields[cpucount]
+				irqdevice = strings.Join(fields[cpucount+1:], " ")
```

Comment:
This could panic if the expected fields are missing.  Since we sliced off the first value, cpucount no longer represents what it is named, I think it needs to be renamed.